h/test/tweens_test.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+
+from pyramid.testing import DummyRequest
+from h import tweens
+
+
+def test_tween_csp_default_headers():
+    request = DummyRequest()
+    tween = tweens.content_security_policy_tween_factory(
+        lambda req: req.response,
+        request.registry)
+
+    response = tween(request)
+
+    assert 'Content-Security-Policy-Report-Only' not in response.headers
+    assert 'Content-Security-Policy' in response.headers
+
+
+def test_tween_csp_report_only_headers():
+    request = DummyRequest()
+    request.registry.settings = {'csp.report_only': True}
+    tween = tweens.content_security_policy_tween_factory(
+        lambda req: req.response,
+        request.registry)
+
+    response = tween(request)
+
+    assert 'Content-Security-Policy-Report-Only' in response.headers
+    assert 'Content-Security-Policy' not in response.headers
+
+
+def test_tween_csp_uri():
+    request = DummyRequest()
+    request.registry.settings = {'csp': {'report-uri': ['localhost']}}
+    tween = tweens.content_security_policy_tween_factory(
+        lambda req: req.response,
+        request.registry)
+
+    response = tween(request)
+
+    assert 'report-uri localhost' in response.headers['Content-Security-Policy']
```

Comment:
It might be good to have a test for the value of the header as well. Just to mitigate future bugs in the code that generates the value string of this header. As far as I understand the user experience could completely break when CSP is enabled and the header value is not what we expect it to be.
The easiest would be to set some specific CSP policy in the test and compare it to the value string we expect it to generate.